test_n13_api.py

```diff
@@ -0,0 +1,46 @@
+from requests import *
+import requests
+
+
+def test_httpbin_post(config):
+    mydata = [{'name': 'Nikolay'}, 'hello']
+    r = post(config['base_url'] + 'post', json=mydata)
+    assert r.status_code == 200, r.text
+    data = r.json()
+    assert data['json'] == mydata
+    assert r.elapsed.total_seconds() < 0.500
+
+
+def test_httpbin_ip(config):
+    r = get(config['base_url'] + 'ip')
+    assert r.status_code == 200, r.text
+    data = r.json()
+    assert data == {u'origin': config['my_ip']}
+    assert r.elapsed.total_seconds() < 0.500
+
+
+def test_user_agent(config):
+    r = get(config['base_url'] + 'user-agent')
+    assert r.status_code == 200, r.text
+    data = r.json()
+    assert data == {u'user-agent': 'python-requests/' + str(requests.__version__)}
+    assert r.elapsed.total_seconds() < 0.500
+
+
+def test_deflate(config):
+    r = get(config['base_url'] + 'deflate')
+    assert r.status_code == 200, r.text
+    data = r.json()
+    assert data['deflated'] == True
+    assert data['headers']['User-Agent'] == 'python-requests/' + str(requests.__version__)
+    assert r.elapsed.total_seconds() < 0.500
+
+
+def test_gzip(config):
+    r = get(config['base_url'] + 'gzip')
+    assert r.status_code == 200, r.text
+    data = r.json()
+    assert data['gzipped'] == True
+    assert data['headers']['User-Agent'] == 'python-requests/' + str(requests.__version__)
+    assert data['headers']['Connection'] == config['connection']
```

Comment:
connection depends on the request header `connection` and if there is a redirection though proxy on the server side. see more  here https://github.com/testlikeachamp/api_tests/pull/6#discussion_r140961233